dotnet/src/Connectors/Connectors.AI.PaLM/Skills/CountTokenSkill.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System;
+using Microsoft.SemanticKernel.SkillDefinition;
+using System.ComponentModel;
+using Microsoft.SemanticKernel.AI;
+using System.Net.Http;
+using Microsoft.SemanticKernel.Diagnostics;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Connectors.AI.PaLM.Skills
+{
+    public sealed class TokenSkill
+    {
+        private const string HttpUserAgent = "Microsoft-Semantic-Kernel";
+
+        private readonly string _model;
+        private readonly string? _endpoint = "https://generativelanguage.googleapis.com/v1beta2/models";
+        private readonly HttpClient _httpClient;
+        private readonly bool _disposeHttpClient = true;
+        private readonly string? _apiKey;
+
+        /// <summary>
+        /// Initializes a new instance of the Token Skill
+        /// </summary>
+        /// <param name="model">Model to use</param>
+        /// <param name="apiKey">PaLM API Key</param>
+        /// <param name="httpClient">instance of http client if already exist</param>
+        public TokenSkill(string model, string apiKey, HttpClient? httpClient = null)
```

Comment:
Make sure it's possible to supply endpoint if needed.

Reply:
fixed